com.intellij.diagnostic.PluginException: Element: class nl.hannahsten.texifyidea.file.LatexFile [Plugin: nl.rubensten.texifyidea]

### Type of JetBrains IDE (IntelliJ, PyCharm, etc.) and version
PyCharm 2021.3.2 (build PC-213.6777.50)

### Operating System 
Mac OS X 12.2.1 (aarch64)

### TeXiFy IDEA version
0.7.15

### Description



### Stacktrace
```
com.intellij.diagnostic.PluginException: Element: class nl.hannahsten.texifyidea.file.LatexFile [Plugin: nl.rubensten.texifyidea]
	at com.intellij.ide.plugins.PluginManagerCore.createPluginException(PluginManagerCore.java:290)
	at com.intellij.diagnostic.PluginProblemReporterImpl.createPluginExceptionByClass(PluginProblemReporterImpl.java:12)
	at com.intellij.diagnostic.PluginException.createByClass(PluginException.java:92)
	at com.intellij.psi.util.PsiUtilCore.ensureValid(PsiUtilCore.java:476)
	at com.intellij.psi.impl.source.PsiFileImpl.loadTreeElement(PsiFileImpl.java:192)
	at com.intellij.psi.impl.source.PsiFileImpl.calcTreeElement(PsiFileImpl.java:734)
	at com.intellij.psi.impl.source.PsiFileImpl.getChildren(PsiFileImpl.java:739)
	at nl.hannahsten.texifyidea.util.PsiCommandsKt.allCommands(PsiCommands.kt:279)
	at nl.hannahsten.texifyidea.util.PsiCommandsKt.allCommands(PsiCommands.kt:208)
	at nl.hannahsten.texifyidea.util.files.PsiFileKt.commandsInFile(PsiFile.kt:209)
	at nl.hannahsten.texifyidea.util.files.PsiFileKt.commandsInFile$default(PsiFile.kt:206)
	at nl.hannahsten.texifyidea.util.files.PsiFileKt.documentClass(PsiFile.kt:70)
	at nl.hannahsten.texifyidea.util.files.PsiFileKt.documentClassFileInProject(PsiFile.kt:64)
	at nl.hannahsten.texifyidea.index.IndexUtilBase.getItemsInFileSet(IndexUtilBase.kt:50)
	at nl.hannahsten.texifyidea.util.files.FileSetKt.commandsInFileSet(FileSet.kt:77)
	at nl.hannahsten.texifyidea.util.PackagesKt.includedPackages(Packages.kt:276)
	at nl.hannahsten.texifyidea.util.PackagesKt.includedPackages$default(Packages.kt:275)
	at nl.hannahsten.texifyidea.reference.InputFileReference.resolve(InputFileReference.kt:136)
	at nl.hannahsten.texifyidea.reference.InputFileReference.resolve$default(InputFileReference.kt:66)
	at nl.hannahsten.texifyidea.reference.InputFileReference.resolve(InputFileReference.kt:47)
	at nl.hannahsten.texifyidea.gutter.LatexNavigationGutter.collectNavigationMarkers(LatexNavigationGutter.kt:54)
	at com.intellij.codeInsight.daemon.RelatedItemLineMarkerProvider.collectNavigationMarkers(RelatedItemLineMarkerProvider.java:31)
	at com.intellij.codeInsight.daemon.RelatedItemLineMarkerProvider.collectSlowLineMarkers(RelatedItemLineMarkerProvider.java:23)
	at com.intellij.codeInsight.daemon.impl.LineMarkersPass.queryProviders(LineMarkersPass.java:195)
	at com.intellij.codeInsight.daemon.impl.LineMarkersPass.lambda$doCollectInformation$3(LineMarkersPass.java:90)
	at com.intellij.codeInsight.daemon.impl.Divider.divideInsideAndOutsideInOneRoot(Divider.java:115)
	at com.intellij.codeInsight.daemon.impl.LineMarkersPass.doCollectInformation(LineMarkersPass.java:85)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:56)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$1(PassExecutorService.java:414)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1084)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:407)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:624)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:698)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:646)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:623)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:66)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:406)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:382)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:174)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:181)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:380)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:184)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: com.intellij.psi.PsiInvalidElementAccessException: Element: class nl.hannahsten.texifyidea.file.LatexFile
	... 44 more
```

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -5,7 +5,7 @@ import org.gradle.api.tasks.testing.logging.TestLogEvent
 // Supersedes the use of "buildscript" block and "apply plugin:"
 plugins {
     id("org.jetbrains.intellij") version "1.4.0"
-    kotlin("jvm") version("1.6.20-RC")
+    kotlin("jvm") version("1.6.20")
 
     // Plugin which can check for Gradle dependencies, use the help/dependencyUpdates task.
     id("com.github.ben-manes.versions") version "0.42.0"

--- a/src/nl/hannahsten/texifyidea/inspections/grazie/LatexTextExtractor.kt
+++ b/src/nl/hannahsten/texifyidea/inspections/grazie/LatexTextExtractor.kt
@@ -24,11 +24,12 @@ class LatexTextExtractor : TextExtractor() {
         // However, we do need it as a root because we need to filter out certain things like inline math ourselves, so that we can make sure all the whitespace around ignored items is correct.
         val domain = TextContent.TextDomain.PLAIN_TEXT
 
-        val textContent = TextContent.builder().build(root, domain)
+        val textContent = TextContent.builder().build(root, domain) ?: return null
         val stealthyRanges = getStealthyRanges(root)
             .map { TextContent.Exclusion.exclude(it.toTextRange()) }
+            .filter { it.start >= 0 && it.end <= textContent.length }
 
-        return textContent?.excludeRanges(stealthyRanges)
+        return textContent.excludeRanges(stealthyRanges)
     }
 
     private fun getStealthyRanges(root: PsiElement): List<IntRange> {

--- a/src/nl/hannahsten/texifyidea/util/PsiCommands.kt
+++ b/src/nl/hannahsten/texifyidea/util/PsiCommands.kt
@@ -203,4 +203,4 @@ fun LatexCommands.forcedFirstRequiredParameterAsCommand(): LatexCommands? {
 /**
  * Get all [LatexCommands] that are children (direct or indirect) of the given element.
  */
-fun PsiElement.allCommands() = childrenOfType(LatexCommands::class)
+fun PsiElement.allCommands() = childrenOfType(LatexCommands::class).toList()

--- a/src/nl/hannahsten/texifyidea/util/PsiCommands.kt
+++ b/src/nl/hannahsten/texifyidea/util/PsiCommands.kt
@@ -201,20 +201,6 @@ fun LatexCommands.forcedFirstRequiredParameterAsCommand(): LatexCommands? {
 }
 
 /**
- * Get all [LatexCommands] that are children of the given element.
+ * Get all [LatexCommands] that are children (direct or indirect) of the given element.
  */
-fun PsiElement.allCommands(): List<LatexCommands> {
-    val commands = ArrayList<LatexCommands>()
-    allCommands(commands)
-    return commands
-}
-
-/**
- * Recursive implementation of [allCommands].
- */
-private fun PsiElement.allCommands(commands: MutableList<LatexCommands>) {
-    forEachChild { it.allCommands(commands) }
-    if (this is LatexCommands) {
-        commands.add(this)
-    }
-}
+fun PsiElement.allCommands() = childrenOfType(LatexCommands::class)

